base16384: bump to version 2.2.4

fumiama opened the pull request:
Maintainer: me
Compile tested: x86/64, Generic PC, OpenWrt 22.03.2 r19803-9a599fee93
Run tested: x86/64, Generic PC, OpenWrt 22.03.2 r19803-9a599fee93

Signed-off-by: 源 文雨 <fumiama@foxmail.com>

1715173329 commented on utils/base16384/test.sh:
```diff
@@ -3,9 +3,9 @@
-            s="$(head /dev/urandom | head -c $i)"
-            if [ "$(echo $s)" != "$(echo $s | base16384 -e - - | base16384 -d - -)" ]; then
+        while [ $i -le 256 ]; do
+            s="$(head /dev/urandom | head -c $i | base64 -w 0)"
```
Command base64 is not included in openwrt by default, you may probably need to install `coreutils-base64` firstly. E.g. insert `opkg install `

fumiama replied:
Thank you for your instruction 🤝. Fixed it.

1715173329 commented:
mips runtime test is failing, is it expected?

fumiama commented:
> mips runtime test is failing, is it expected?

It is not what I had expected and it seems to happen after I added `opkg install`.

When I open this PR with only a version changing, I did get this mips arch error. So I added base64 to test whether it is caused by the binary-echo issue, then all test passed so a think it has been fixed. However, after I added `opkg install`, it failed again.

Here are two tests in my repo.

1. https://github.com/fumiama/packages/pull/2 mips failed
2. https://github.com/fumiama/packages/pull/1 all tests passed

1715173329 commented:
> 2. [test: without opkg install fumiama/packages#1](https://github.com/fumiama/packages/pull/1) all tests passed

It's not really "passed" as in logs it prints "`/ci/utils/base16384/test.sh: line 7: base64: not found`"

fumiama commented:
Sorry, I read the log again and found that the mips test passed because base64 not found. So it seems to be the program's own issue. I will do more test and try to fix it soon.

fumiama commented:
Thanks for your instruction, I have detected a bug and released a new version 2.2.4. Hope no problem this time 😂.

1715173329 commented:
Good job, thanks!

1715173329 commented:
would you mind me cherry-picking (backporting) this commit to the 23.05 branch?

fumiama commented:
> would you mind me cherry-picking (backporting) this commit to the 23.05 branch?

No, I'd like it.

1715173329 commented:
done in fb4b7a4aaaf668d958754aab4d052628d309250e ;)
